osu.Game/Overlays/BeatmapSet/Buttons/FavouriteButton.cs

```diff
@@ -67,16 +53,22 @@ private void load()
 
             favourited.ValueChanged += favourited =>
             {
-                if (favourited.NewValue)
-                {
-                    pink.FadeIn(200);
-                    icon.Icon = FontAwesome.Solid.Heart;
-                }
-                else
-                {
-                    pink.FadeOut(200);
-                    icon.Icon = FontAwesome.Regular.Heart;
-                }
+                loading.Hide();
+
+                icon.Icon = favourited.NewValue ? FontAwesome.Solid.Heart : FontAwesome.Regular.Heart;
+            };
+
+            Action = () =>
+            {
+                if (loading.State.Value == Visibility.Visible)
+                    return;
+
+                loading.Show();
+
+                request?.Cancel();
+                request = new PostBeatmapFavouriteRequest(BeatmapSet.Value?.OnlineBeatmapSetID ?? 0, favourited.Value ? BeatmapFavouriteAction.UnFavourite : BeatmapFavouriteAction.Favourite);
+                request.Success += () => favourited.Value = !favourited.Value;
```

Comment:
I'm a bit worried about this potentially desyncing with the actual web state. Does that endpoint not actually return whether or not the favourite was added or removed? I'd be less concerned about potential discrepancies then.

Reply:
afaik it returns total favourite beatmaps amount

Reply:
Agh, that doesn't help here much. I guess this is passable then.

Reply:
I agree with @bdach, you should open an issue web-side for this, this may be dangerous.

Reply:
Yep, should ask for API change.